strawberryfields/api/devicespec.py

```diff
@@ -100,25 +104,7 @@ def gate_parameters(self):
 
         return gate_parameters
 
-    def layout_is_formatted(self):
-        """bool: Whether the device layout is formatted or not."""
-        p = re.compile(r"{{\w*}}")
-        return not bool(p.search(self.layout))
-
-    def fill_template(self, program):
-        """Fill template with parameter values from a program"""
-        if self.layout_is_formatted():
-            return
-
-        if isinstance(program, TDMProgram):
-            self._spec["layout"] = self._spec["layout"].format(
-                target=self.target, tm=program.timebins
-            )
-        else:
-            # TODO: update when `self._spec["layout"]` is returned as an unformatted string
-            raise NotImplementedError("Formatting not required or supported for non-TDM programs.")
```

Comment:
Just to confirm, these are no longer needed, right?

Reply:
Shouldn't be. They were only used when the TDM spec was returning an unformatted layout, which is no longer the case. These functions aren't called anywhere in neither SF nor OG.